lib/portage/package/ebuild/doebuild.py

```diff
@@ -486,7 +486,10 @@ def doebuild_environment(myebuild, mydo, myroot=None, settings=None,
 		distcc = "distcc" in mysettings.features
 		icecream = "icecream" in mysettings.features
 
-		if ccache or distcc or icecream:
+		# run cc thingies only in src_* phases as otherwise they might
+		# create files with incorrect ownership and trip over
+		if (ccache or distcc or icecream) and mydo in ('unpack',
+				'prepare', 'configure', 'test', 'install'):
```

Comment:
The `install` phase is different from the others because it runs as root. Is that ok?

Reply:
I've decided to include it because 1) there are some bad packages that recompile stuff in `src_install`, and then ccache is very helpful, and 2) almost always some compilation happens before `src_install`, so permissions are set up already. Do you want me to add an explanatory comment there?

Reply:
Yeah, a comment would be great. Thanks!